src/detect-content.c

```diff
@@ -3050,6 +3056,22 @@ static int DetectLongContentTest3(void)
     return !DetectLongContentTestCommon(sig, 1);
 }
 
+static int DetectBadBinContent(void)
+{
+    DetectEngineCtx *de_ctx = NULL;
+    de_ctx = DetectEngineCtxInit();
+    FAIL_IF_NULL(de_ctx);
+    de_ctx->flags |= DE_QUIET;
+    FAIL_IF_NOT_NULL(SigInit(
```

Comment:
Prefer `DetectEngineAppendSig` as this use of `SigInit` will leak memory in the success path.